src/Agent.Worker/TaskRunner.cs

```diff
@@ -525,5 +538,24 @@ private void PrintTaskMetaData(Definition taskDefinition)
             ExecutionContext.Output($"Help         : {taskDefinition.Data.HelpUrl ?? taskDefinition.Data.HelpMarkDown}");
             ExecutionContext.Output("==============================================================================");
         }
+
+        private bool IsInputContainsSecret(string inputValue)
+        {
+            string maskedString = HostContext.SecretMasker.MaskSecrets(inputValue);
+            return maskedString.Contains("***");
```

Comment:
Nit: this creates indirect duplication (you rely on fact that secrets masked by ***) - could there be any more generic way to check it? Could inputValue != maskedString condition be more generic here?